tracer/src/Datadog.Trace/Logging/TracerFlare/TracerFlareManager.cs

```diff
@@ -0,0 +1,327 @@
+            var result = new ApplyDetails[config.Count];
+            for (var i = 0; i < result.Length; i++)
+            {
+                var remoteConfig = config[i];
+                result[i] = TrySendDebugLogs(remoteConfig.Path.Path, remoteConfig.Contents, remoteConfig.Path.Id, fileLogDirectory);
+            }
+
+            return result;
+        }
+        catch (Exception ex)
+        {
+            Log.Error(ex, "Error handling tracer flare request");
+
+            return ErrorAll(config, ex);
+        }
+    }
+
+    // internal for testing
+    internal ApplyDetails TrySendDebugLogs(string configPath, byte[] configContents, string configId, string fileLogDirectory)
+    {
+        try
+        {
+            // Looking for config that looks like this:
+            // { "task_type": "tracer_flare", "args" { "case_id": "12345"} }
+            var jObject = TryDeserialize(configContents);
+            if (jObject is null)
+            {
+                Log.Debug("Invalid configuration provided for tracer flare");
+                return ApplyDetails.FromError(configPath, "Invalid configuration provided");
+            }
+
+            if (!jObject.TryGetValue("task_type", StringComparison.Ordinal, out var taskTypeToken)
+             || taskTypeToken.Type != JTokenType.String
+             || !string.Equals(taskTypeToken.Value<string>(), "tracer_flare", StringComparison.Ordinal))
+            {
+                // not the right sort of task, just acknowledge it
+                Log.Debug($"{RcmProducts.TracerFlareRequested} was not tracer flare - ignoring");
+                return ApplyDetails.FromOk(configPath);
+            }
+
+            if (!jObject.TryGetValue("args", StringComparison.Ordinal, out var args)
+             || args is not JObject argsObject
+             || !argsObject.TryGetValue("case_id", out var caseIdToken)
+             || caseIdToken.Type != JTokenType.String
+             || caseIdToken.Value<string>() is not { Length: > 0 } caseId)
+            {
+                // missing case_id - should we just accept it?
+                Log.Debug("Invalid configuration provided for tracer flare - missing case_id");
+                return ApplyDetails.FromError(configPath, "Missing case_id");
+            }
+
+            if (!DebugLogReader.TryToCreateSentinelFile(fileLogDirectory, configId))
+            {
+                // already accepted by a different tracer
+                Log.Debug("Tracer flare already handled by other tracer instance - ignoring");
+                return ApplyDetails.FromOk(configPath);
+            }
+
+            // ok, do the thing
+            var task = _flareApi.SendTracerFlare(stream => DebugLogReader.WriteDebugLogArchiveToStream(stream, fileLogDirectory), caseId);
+
+            // uh oh, sync over async :grimace:
+            // and this all happens inside a lock IIRC, so this could be problematic...
+            var result = task.GetAwaiter().GetResult();
```

Comment:
FYI, @kevingosse is working on making the RC system async, so can hopefully remove this 🤞

Reply:
Kevin fixed in #4996, so no sync-over-async now 🙌